cmake/kokkos_arch.cmake

```diff
@@ -389,11 +389,11 @@ IF(KOKKOS_ARCH_${ARCH})
     MESSAGE(FATAL_ERROR "Multiple GPU architectures given! Already have ${CUDA_ARCH_ALREADY_SPECIFIED}, but trying to add ${ARCH}. If you are re-running CMake, try clearing the cache and running again.")
   ENDIF()
   SET(CUDA_ARCH_ALREADY_SPECIFIED ${ARCH} PARENT_SCOPE)
-  GLOBAL_APPEND(KOKKOS_CUDA_OPTIONS "${CUDA_ARCH_FLAG}=${FLAG}")
-  GLOBAL_APPEND(KOKKOS_LINK_OPTIONS "${CUDA_ARCH_FLAG}=${FLAG}")
```

Comment:
We need this for `KOKKOS_ENABLE_RELOCATABLE_DEVICE_CODE=ON`, see #2400.

Reply:
@dalg24 @jjwilke @crtrott @masterleinad 
Add link options back in for relocatable device code enabled.  We will sort out the Clang linking problem with a different issue.  This shouldn't affect any existing tests.